### PR TITLE
Update drawer/bottom-actions component

### LINKED ISSUE #15006: Quo 2 - Update Drawers / Bottom Actions component to adapt the new design
Update this component in the quo2 component library as per the design spec:

Designs - Drawers / Bottom Actions
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=2638%3A35963&t=29WaMjPTYMtHqa1n-0

** There is an implementation of this more or less done in https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/communities/menus/request_to_join/view.cljs#L112-L132

Acceptance Criteria
- dark and light mode
- component tests (jest)
- quo2 preview
- export from quo2/core.cljs
- adjust usage in communities join screen to use this component

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #15006 

This PR updates the drawer/bottom-actions component(Quo Preview -> Drawer -> Bottom Actions) with the new options. [Check the design for details.](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=2637-35934&mode=dev)
- Top

- Top error

### Other fix
This PR also addresses a minor bug in the button component. The bug was causing the override of the type-based background color when using customization-color in conjunction with the `positive` or `danger` buttons.

status: ready

### CODE REVIEW

File: src/quo/components/buttons/button/view.cljs [hunk 1]
```diff
@@ -37,14 +37,16 @@
...
-                                         (= type :danger)   :danger
-                                         :else              nil)}}
+               customization-color :blue}}
        children]
-      (let [{:keys [icon-color background-color label-color border-color blur-type
+      (let [customization-color-by-type (condp = type
+                                          :primary  customization-color
+                                          :positive :success
+                                          :danger   :danger
+                                          nil)
```
Comment (smohamedjavid) [inline src/quo/components/buttons/button/view.cljs]: IMO, we should fix this in the `properties.cljs` as customization-color is not applicable when it's not type `primary`. WDYT?

https://github.com/status-im/status-mobile/blob/61c0650ceca1d0d6f44fedfd34ccc34f8fe99e10/src/quo/components/buttons/button/properties.cljs#L81-L86

cc: @J-Son89

File: src/quo/components/buttons/button/view.cljs [hunk 2]
```diff
@@ -78,9 +80,9 @@
                                             :icon-left        icon-left
                                             :icon-right       icon-right})
                     inner-style)}
-           (when customization-color
+           (when customization-color-by-type
```
Comment (smohamedjavid) [inline src/quo/components/buttons/button/view.cljs]: We can add a check on whether the type is `primary`.

File: src/quo/components/buttons/button/view.cljs [hunk 3]
```diff
@@ -78,9 +75,12 @@
                                             :icon-left        icon-left
                                             :icon-right       icon-right})
                     inner-style)}
-           (when customization-color
+           (when (contains? #{:primary :positive :danger} type)
              [customization-colors/overlay
-              {:customization-color customization-color
+              {:customization-color (get {:primary  customization-color
+                                          :positive :success
+                                          :danger   :danger}
```
Comment (smohamedjavid) [inline src/quo/components/buttons/button/view.cljs]: We can move this to `properties.cljs` for better maintainability.

**Review by smohamedjavid — APPROVED** [review]
Nice work! 🙌

**Review by OmarBasem — APPROVED** [review]

**Comment by pavloburykh** [comment]
@ajayesivan thanx for the PR. No issues from my side. Frankly speaking I have not noticed any visual difference of those buttons comparing to what is now in develop. But I might definitely be missing something. 

I believe we need design review from @Francesca-G here.

**Review by Francesca-G — COMMENTED** [review]
Here's the [design review](https://www.figma.com/file/e9woQowfaIOCBrXS1jgkhZ/Review-Components-%2F-Mobile?type=design&node-id=4%3A34&mode=design&t=ZifEFuVtOWcPT6Lm-1) :)

**Comment by ajayesivan** [comment]
@Francesca-G Thank you for the design review. I have updated the PR with design fixes, could you please have another look?

**Review by Francesca-G — APPROVED** [review]
Looks good now ✨

**Comment by ajayesivan** [comment]
Hi @pavloburykh, thank you for testing the PR. Could you please have a look at the e2e results?

**Comment by pavloburykh** [comment]
> Hi @pavloburykh, thank you for testing the PR. Could you please have a look at the e2e results?

Hi @ajayesivan. Failed e2e are not related to PR. Most failures are related to data delivery issues. Ready for merge.

**Comment by OmarBasem** [comment]
Hi @ajayesivan, I think this PR introcuded two issues to the bottom action component. 

1. On some screens the bottom action component causes a white overlay on the screen
2. Container-style prop doesn't seem to work now. 
Screenshot_20240212_093531_Status

File: src/quo/components/drawers/bottom_actions/view.cljs [hunk 4]
```diff
@@ -2,55 +2,105 @@
...
+  [{:keys [actions description description-text error-message role button-one-label button-two-label
+           blur? button-one-props button-two-props theme scroll? container-style]}]
+  [blur/view
+   {:blur-amount   20
+    :blur-type     :transparent
+    :overlay-color (if blur?
+                     colors/neutral-80-opa-1-blur
+                     (colors/theme-colors colors/white-70-blur
+                                          colors/neutral-95-opa-70-blur
+                                          theme))}
```
Comment (OmarBasem) [inline src/quo/components/drawers/bottom_actions/view.cljs:51]: @ajayesivan I think this is the problem. Probably wrapping with blur view should be conditional

Comment (ajayesivan) [reply src/quo/components/drawers/bottom_actions/view.cljs:51]: @OmarBasem Thanks for reporting this. I will look into it.

Comment (ajayesivan) [reply src/quo/components/drawers/bottom_actions/view.cljs:51]: @OmarBasem, I recall that @smohamedjavid addressed this issue in one of his PRs, and I couldn't reproduce this issue in the latest develop. Could you confirm if the problem persists in the latest develop branch? If it does, please specify which screens are still experiencing this issue. Thank you!

**Pull request merged**